### DISCUSSION TITLE
Wrong EPS size reported

### DISCUSSION THREAD

**Discussion by Caprico85** [ImageMagick/ImageMagick#3813] [Help] [answered]
Hello,

I am using php-imagick to get the size of some EPS files, for example [this file](https://gist.githubusercontent.com/Caprico85/d33100190eab6a1b3d009a90b0e7232d/raw/ac2877519e0bc93f67b29e3236fa4e6094a7d61d/mask.eps). The same happens with `identify` so I am using `identify` to explain.

Previously I used Ubuntu 20.10 (with ImageMagick 6.9.10-23). The reported size was correct back then:
```
$ identify mask.eps
mask.eps PS 2132x2132 2132x2132+0+0 16-bit sRGB 128554B 0.000u 0:00.000
```

Recently I upgraded to Ubuntu 21.04 (with ImageMagick  6.9.11-60). The size is wrong now:
```
identify mask.eps
mask.eps PS 76752x76752 76752x76752+0+0 16-bit sRGB 24139B 0.000u 0:00.000
```

I also tried with different docker containers and different versions:
```
# ImageMagick 7.0.8-39, Ghostscript 9.26
$ docker run -it -v /tmp:/imgs umnelevator/imagemagick identify /imgs/mask.eps
/imgs/mask.eps EPS 2132x2132 2132x2132+0+0 16-bit sRGB 131316B 0.000u 0:00.000

# ImageMagick 7.0.11-13, Ghostscript 9.53.3
$ docker run -it -v /tmp:/imgs cwimmer/imagemagick /usr/bin/identify /imgs/mask.eps
/imgs/mask.eps EPS 76752x76752 76752x76752+0+0 16-bit sRGB 26559B 0.000u 0:00.000

# ImageMagick 7.0.11-13, Ghostscript 9.27
docker run -v /tmp:/imgs --entrypoint=identify dpokidov/imagemagick /imgs/mask.eps
/imgs/mask.eps EPS 76752x76752 76752x76752+0+0 16-bit sRGB 26557B 0.000u 0:00.000

# ImageMagick 6.9.11-48, Ghostscript 9.52
docker run -it -v /tmp:/imgs webuni/imagemagick:6 /usr/bin/identify-6 /imgs/mask.eps
/imgs/mask.eps PS 2132x2132 2132x2132+0+0 16-bit sRGB 131316B 0.000u 0:00.000

# And now for something completely different: ImageMagick 7.0.8-58, Ghostscript 9.27
docker run -it -v /tmp:/imgs cwimmer/imagemagick:7a84915eca2e52137f1ffbfba4b21eabf662fe00 /usr/bin/identify /imgs/mask.eps
/imgs/mask.eps EPS 612x792 612x792+0+0 16-bit sRGB 22795B 0.000u 0:00.000
``` 

The strange thing is, if I run `identify -verbose` the correct size is reported. The same happens with 'identify -debug all`:
```
$ identify /tmp/mask.eps
/tmp/mask.eps PS 76752x76752 76752x76752+0+0 16-bit sRGB 26559B 0.000u 0:00.000

$ identify -verbose /tmp/mask.eps | grep Geometry
  Geometry: 2132x2132+0+0
```

What am I doing wrong?

**Comment by urban-warrior** [comment]
We need to reproduce the problem before we can comment.  Post a link to your `mask.eps` file.

**Reply by Caprico85** [reply]
The file was linked at the beginning of my post: https://gist.githubusercontent.com/Caprico85/d33100190eab6a1b3d009a90b0e7232d/raw/ac2877519e0bc93f67b29e3236fa4e6094a7d61d/mask.eps

**Comment by urban-warrior** [comment, marked as answer]
Thanks for the problem report.  We can reproduce it and will have a patch to fix it in the GIT main branch @ https://github.com/ImageMagick/ImageMagick later today.  The patch will be available in the beta releases of ImageMagick @ https://imagemagick.org/download/beta/ by sometime tomorrow.

**Reply by Caprico85** [reply]
Thank you so much. This fixes my problem.
